src/hooks/http.ts

```diff
@@ -48,15 +48,23 @@ export class Http {
       GlobalDurationTimer.start();
       if (isEmptyString(requestData.body)) {
         const body = extractBodyFromWriteOrEndFunc(args);
-        if (body) {
-          requestData.body += body.substr(0, getEventEntitySize(false));
-        }
-        if (currentSpan) currentSpan.info.httpInfo = getHttpInfo(requestData, {});
+        Http.aggregateRequestBodyToSpan(body, requestData, currentSpan);
       }
       GlobalDurationTimer.stop();
     };
   }
 
+  static aggregateRequestBodyToSpan(body, requestData, currentSpan) {
```

Comment:
please add tests to every aspect of this function.
It is extremely complex, so it should be tested deeply